pkgs/by-name/co/colorstorm/package.nix

```diff
@@ -2,30 +2,39 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  zig_0_9,
+  zig_0_13,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "colorstorm";
-  version = "2.0.0";
+  # last tagged release is three years old and requires outdated Zig 0.9
+  # new release requested in: https://github.com/benbusby/colorstorm/issues/16
+  version = "2.0.0-unstable-2025-01-17";
 
   src = fetchFromGitHub {
     owner = "benbusby";
     repo = "colorstorm";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-6+P+QQpP1jxsydqhVrZkjl1gaqNcx4kS2994hOBhtu8=";
+    rev = "e645c4293fb5f72968038dac99e0b8dab3db194f";
+    hash = "sha256-6D+aNcjJksv7E9RJB9fnzgzvGoUPXV4Shz5wLu5YHtg=";
   };
 
+  patches = [
+    # Fixes a use-after-free segfault.
+    # See https://github.com/benbusby/colorstorm/pull/15#discussion_r1930406581
+    # and upstream PR https://github.com/NixOS/nixpkgs/pull/377279
+    ./0001-fix-segfault.patch
```

Comment:
can we use fetchpatch instead ?

Reply:
Usually we only use fetchpatch if the patch has actually been merged in upstream

Reply:
it depends on how critical/big project is, in this case I would say adding a file to nixpkgs repo is not worth it

Reply:
On the contrary, adding small patches to nixpkgs is without much consequence and also we don't even know if the upstream PR will merged as-is and the commit the patch would then reference might disappear. Resolving this discussion.